applicationset/controllers/applicationset_controller.go

```diff
@@ -1430,7 +1430,8 @@ func syncApplication(application argov1alpha1.Application, prune bool) argov1alp
 				Value: "ApplicationSet RollingSync triggered a sync of this Application resource.",
 			},
 		},
-		Sync: &argov1alpha1.SyncOperation{},
+		Sync:  &argov1alpha1.SyncOperation{},
+		Retry: argov1alpha1.RetryStrategy{Limit: 5},
```

Comment:
Hardcoding this doesn't seem right. I believe the retry policy for apps should be derived from the app configuration, but correct me if I'm wrong.